src/shared/parameter_v2/config/example.yaml

```diff
@@ -0,0 +1,75 @@
+# Examples of all supported Parameter types and their schemas
+--- # Metadata for this config
+metadata:
+  # If shared is true, all parameters in this file will be visible in configs at the same
+  # level in the directoy structure, as well as all children.
+  shared: false
+  # The name of this config. This will be the name used in the generated code
+  name: ExampleConfig
+--- # Parameters
+BoolParam:
+  type: bool
+  value: true
+  description: >
+    Can be true or false
+DoubleParam:
+  type: double # could also be float
+  min: 0
+  max: 5
+  value: 0.2
+  description: >
+    Can be any value in the range [min, max]
+IntParam:
+  type: int # could also be unsigned
+  min: 0
+  max: 5
+  value: 3
+  description: >
+    Can be any integer value in the range [min, max]
+StringParam:
+  type: std::string
+  value: "Hello World"
+  description: >
+    Any string value. Will NEVER be generated for C code
```

Comment:
Technically we could generate strings for C, they would just be `const char*`, which implies the `const` attribute in the YAML.

I really don't think we will ever need string constants in C, but at the same time it's generally not good to have this one-off conditional behavior in schemas. Thoughts on how to make this more intuitive for those writing configs?

Edit: Use case for strings in C is IP addresses, so this is valid.

Reply:
Updated the string param